src/azure-cli/azure/cli/command_modules/sqlvm/_validators.py

```diff
@@ -97,3 +97,14 @@ def validate_subnet(cmd, namespace):
             name=vnet, child_type_1='subnets',
             child_name_1=subnet
         )
+
+
+# pylint: disable=too-many-statements,line-too-long
+def validate_sqlmanagement(namespace):
+    '''
+    Validates if sql management mode provided, the offer type and sku type has to be provided.
+    '''
+    sql_mgmt_mode = namespace.sql_management_mode
+
+    if (sql_mgmt_mode == "NoAgent" and (namespace.sql_image_sku is None or namespace.sql_image_offer is None)):
+        raise CLIError("incorrect usage: --sql-mgmt-type NoAgent --image-sku NAME --image-offer NAME")
```

Comment:
Can we explicitly mention that for NoAgent, the only available Image is SQL2008-WS2008 and SQL2008R2-WS2008?

Reply:
We do it, in the examples itself (_help.py). This validation is just for usage and should not have any examples.